inception/inception-api-annotation-experimental/src/main/ts/client/AnnotationExperienceAPIImpl.ts

```diff
@@ -16,293 +16,448 @@
+                    j, function (msg) {
+                    that.onSpanUpdate(Object.assign(new UpdateSpanResponse(), JSON.parse(msg.body)));
+                }, {id: "span_update_" + j});
+
+                this._stompClient.subscribe("/topic/span_create_for_clients/" +
+                    this._projectID + "/" +
+                    this._documentID + "/" +
+                    j, function (msg) {
+                    that.onSpanCreate(Object.assign(new CreateSpanResponse(), JSON.parse(msg.body)));
+                }, {id: "span_create_" + j});
+
+                this._stompClient.subscribe("/topic/span_delete_for_clients/" +
+                    this._projectID + "/" +
+                    this._documentID + "/" +
+                    j, function (msg) {
+                    that.onSpanDelete(Object.assign(new DeleteSpanResponse(), JSON.parse(msg.body)));
+                }, {id: "span_delete_" + j});
+
+                this._stompClient.subscribe("/topic/relation_update_for_clients/" +
+                    this._projectID + "/" +
+                    this._documentID + "/" +
+                    j, function (msg) {
+                    that.onRelationUpdate(Object.assign(new UpdateRelationResponse(), JSON.parse(msg.body)));
+                }, {id: "relation_update_" + j});
+
+                this._stompClient.subscribe("/topic/relation_create_for_clients/" +
+                    this._projectID + "/" +
+                    this._documentID + "/" +
+                    j, function (msg) {
+                    that.onRelationCreate(Object.assign(new CreateRelationResponse(), JSON.parse(msg.body)));
+                }, {id: "relation_create_" + j});
+
+                this._stompClient.subscribe("/topic/relation_delete_for_clients/" +
+                    this._projectID + "/" +
+                    this._documentID + "/" +
+                    j, function (msg) {
+                    that.onRelationDelete(Object.assign(new DeleteRelationResponse(), JSON.parse(msg.body)));
+                }, {id: "relation_delete_" + j});
+            }
+        }
     }
 
     unsubscribe(aChannel: string) {
-        this.stompClient.unsubscribe(aChannel);
+        this._stompClient.unsubscribe(aChannel);
     }
 
-    // ------ DISCONNECT -------- //
     disconnect() {
-        if (this.connected) {
-            console.log("Disconnecting now");
-            this.connected = false;
-            this.stompClient.deactivate();
-        }
+        this._stompClient.deactivate();
     }
 
-    editAnnotation(aId, aAnnotationType) {
-        let json = {
-            username: this.client,
-            clientName: this.client,
-            project: this.projectID,
-            document: this.documentID,
-            annotationAddress: aId,
-            annotationType: aAnnotationType
-        };
+    requestNewDocumentFromServer(aClientName: string, aUserName: string, aProjectId: number, aDocumentId: number, aViewport: Viewport) {
```

Comment:
"new document" sounds to me like creating a new document. I think you mean "request-document" or maybe even "subscribe-to-document"?. Why the "from server" here in the call and "from client" in the url? Isn't it obvious that one is the server and the other is the client? Isn't the subscription "to document" and "to viewport" redundant? If you subscribe to a viewport that spans an entire document, then it is like "subscribe to document", right? If yes, why two endpoints?

Reply:
If you want to just quickly write an annotation editor without wanting to worry about paging and/or dynamic-loading-scrolling, you'd just subscribe to the whole document as a viewport. Probably only after doing that would one go into optimizing the editor by introducing some paging or dynamic-loading-scrolling mechanism - just because building that is additional work.